nullaway/src/main/java/com/uber/nullaway/handlers/contract/fieldcontract/RequiresNonNullHandler.java

```diff
@@ -0,0 +1,212 @@
+
+  /** All methods can add the precondition of {@code RequiresNonNull}. */
+  @Override
+  protected boolean validateAnnotationSemantics(
+      NullAway analysis, VisitorState state, MethodTree tree, Symbol.MethodSymbol methodSymbol) {
+    return true;
+  }
+
+  /**
+   * All overriding methods can only weaken the precondition of its super method. No overriding
+   * methods can add new field names to the set of fields of its super method mentioned in {@code
+   * RequiresNonNull} annotation.
+   */
+  @Override
+  protected void validateOverridingRules(
+      Set<String> overridingFieldNames,
+      NullAway analysis,
+      VisitorState state,
+      MethodTree tree,
+      Symbol.MethodSymbol overriddenMethod) {
+    if (overridingFieldNames == null) {
+      return;
+    }
+    Set<String> overriddenFieldNames = getAnnotationValueArray(overriddenMethod, annotName, false);
+    if (overriddenFieldNames == null) {
+      overriddenFieldNames = Collections.emptySet();
+    }
+    overriddenFieldNames = ContractUtils.trimReceivers(overriddenFieldNames);
+    if (overriddenFieldNames.containsAll(overridingFieldNames)) {
+      return;
+    }
+    overridingFieldNames.removeAll(overriddenFieldNames);
+    StringBuilder errorMessage = new StringBuilder();
+    errorMessage.append(
+        "precondition inheritance is violated, method in child class cannot have a stricter precondition than its closest overridden method, adding @requiresNonNull for fields [");
+    Iterator<String> iterator = overriddenFieldNames.iterator();
+    while (iterator.hasNext()) {
+      errorMessage.append(iterator.next());
+      if (iterator.hasNext()) {
+        errorMessage.append(", ");
+      }
+    }
+    errorMessage.append("] makes this method precondition stricter");
+
+    state.reportMatch(
+        analysis
+            .getErrorBuilder()
+            .createErrorDescription(
+                new ErrorMessage(
+                    ErrorMessage.MessageTypes.WRONG_OVERRIDE_PRECONDITION, errorMessage.toString()),
+                tree,
+                buildDescriptionFromChecker(tree, analysis),
+                state));
+  }
+
+  /**
+   * It checks whether all class fields given in {@code RequiresNonNull} parameter are @NonNull at
+   * call site.
+   */
+  @Override
+  public void onMatchMethodInvocation(
+      NullAway analysis,
+      MethodInvocationTree tree,
+      VisitorState state,
+      Symbol.MethodSymbol methodSymbol) {
+    Set<String> fieldNames = getAnnotationValueArray(methodSymbol, annotName, false);
+    if (fieldNames == null) {
+      super.onMatchMethodInvocation(analysis, tree, state, methodSymbol);
+      return;
+    }
+    fieldNames = ContractUtils.trimReceivers(fieldNames);
+    for (String fieldName : fieldNames) {
+      Symbol.ClassSymbol classSymbol = ASTHelpers.enclosingClass(methodSymbol);
+      Preconditions.checkNotNull(
+          classSymbol, "can not find the enclosing class for method symbol: " + methodSymbol);
```

Comment:
`can not` and `cannot` are not equivalent and this one should be `cannot find` ("is unable to find...", rather than "it might not find..." :) )

Reply:
https://github.com/uber/NullAway/pull/423/commits/e75b6a22dc2c7068085a62a4b20bce234254d676